go/arrow/flight/flightsql/driver/utils.go

```diff
@@ -60,44 +60,135 @@ func (g grpcCredentials) RequireTransportSecurity() bool {
 func fromArrowType(arr arrow.Array, idx int) (interface{}, error) {
 	switch c := arr.(type) {
 	case *array.Boolean:
+		if c.IsNull(idx) {
```

Comment:
because `IsNull` is in the `arrow.Array` interface, you can probably factor this out and just do `if arr.IsNull(idx) { return nil, nil }` before the typeswitch both to simplify the code and to create a fast path for nulls to avoid the type switch entirely in that case.